File: exercises/algorithm-patterms/warmup/contains-duplicates/problem.js
/*
 * @param {array} nums
 * @return {boolean} true if there are duplicates, false if not
 *
 * time complexity: O(n)
 * space complexity: O(n)
 */
class Solution {
   containsDuplicate(nums) {
      const duplicates = new Set()

      for (const num of nums) {
         if (duplicates.has(num)) {
            return true
         }
         duplicates.add(num)
      }

      return false
   }
}

module.exports = Solution

File: exercises/algorithm-patterms/warmup/contains-duplicates/problem.py
class Solution:
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        duplicates = set()

        for num in nums:
            if num in duplicates:
                return True
            else:
                duplicates.add(num)

        return False
